mixins/demo/visibility-test.js

```diff
@@ -1,19 +1,51 @@
+import '../../components/button/button.js';
+import './visibility-test-list.js';
 import { css, html, LitElement } from 'lit-element/lit-element.js';
-import { VisibilityMixin } from '../visibility-mixin.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
 
-export class VisibilityTest extends VisibilityMixin(LitElement) {
+export class VisibilityTest extends LitElement {
+
+	static get properties() {
+		return {
+			_listVisibility: { type: Boolean, reflect: false }
+		};
+	}
 
 	static get styles() {
 		return css`
 			:host {
 				display: block;
+				max-width: 400px;
 			}
 		`;
 	}
 
+	constructor() {
+		super();
+		this._listVisibilityChanged = false;
+		this._listVisibility = true;
+	}
+
 	render() {
-		return html`<div>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</div>`;
+		const animateValue = this._listVisibilityChanged ?
+			(this._listVisibility ? 'show' : 'hide') : undefined;
+		const toggleText = this._listVisibility ? 'Hide List' : 'Show List';
+		return html`
+			<d2l-visibility-test-list animate="${ifDefined(animateValue)}"></d2l-visibility-test-list>
+			<d2l-button @click="${this._handleAddItem}">Add Item</d2l-button>
+			<d2l-button @click="${this._handleToggleList}">${toggleText}</d2l-button>
```

Comment:
Should the `Add Item` button be disabled while the list is hidden? Adding while the list is hidden leaves a new dummy with `height: 0`, wrapped around a new item with the next number.

Reply:
Agreed -- fixed!